### PR TITLE
fix: deleteClient crash on missing managedObjectContext

### PR DESCRIPTION
**Pull request by netbe**
### Issue

Fix crash (nil found) on debug builds where a client is deleted

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

### CODE REVIEW

**Review by markusfassbender — APPROVED** [review]
Is it possible to reproduce those steps for testing?

**Comment by netbe** [comment]
> Is it possible to reproduce those steps for testing?

@markusfassbender not sure it's second time it happens to me in different times, I'll check

File: wire-ios-data-model/Source/Model/UserClient/UserClient.swift [hunk 1]
```diff
@@ -265,7 +265,9 @@ public class UserClient: ZMManagedObject, UserClientType {
     }
 
     private func deleteClient() {
-        assert(self.managedObjectContext!.zm_isSyncContext, "clients can only be deleted on syncContext")
+        guard let managedObjectContext else { return }
+
+        assert(managedObjectContext.zm_isSyncContext, "clients can only be deleted on syncContext")
```
Comment (caldrian) [inline wire-ios-data-model/Source/Model/UserClient/UserClient.swift:270]: I had this crash too, but couldn't reproduce it.
But I wonder, how could the `managedObjectContext` become `nil` while/before executing the `managedObjectContext.perform` closure?

Comment (caldrian) [reply wire-ios-data-model/Source/Model/UserClient/UserClient.swift:270]: > I wonder, how could the managedObjectContext become nil while/before executing the managedObjectContext.perform closure?

If that's really the case, what about holding a strong reference on the caller side?
```
if let managedObjectContext {
    await managedObjectContext.perform { self.deleteClient() }
}
```

Comment (netbe) [reply wire-ios-data-model/Source/Model/UserClient/UserClient.swift:270]: it seems to happen when i change branch

Comment (netbe) [reply wire-ios-data-model/Source/Model/UserClient/UserClient.swift:270]: the nil part is either we don't have the coredatastack or the userclient was deleted already therefore not attached to the MOC

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**